service/src/main/resources/api/service_openapi.yaml

```diff
@@ -994,6 +1093,110 @@ components:
+          items:
+            $ref: "#/components/schemas/CriteriaGroupV2"
+
+    CohortListV2:
+      type: array
+      items:
+        $ref: "#/components/schemas/CohortV2"
+
+    CohortCreateInfoV2:
+      type: object
+      properties:
+        underlayName:
+          $ref: "#/components/schemas/UnderlayNameV2"
+        displayName:
+          $ref: "#/components/schemas/CohortDisplayNameV2"
+        description:
+          $ref: "#/components/schemas/CohortDescriptionV2"
+
+    CohortUpdateInfoV2:
+      type: object
+      properties:
+        displayName:
+          $ref: "#/components/schemas/CohortDisplayNameV2"
+        description:
+          $ref: "#/components/schemas/CohortDescriptionV2"
+        criteriaGroups:
+          type: array
+          description: Groups of criteria that define the cohort
+          items:
+            $ref: "#/components/schemas/CriteriaGroupV2"
+
+    UnderlayNameV2:
+      type: string
+      description: Name of the underlay, immutable
+
+    CohortDisplayNameV2:
+      type: string
+      description: Human readable name of the cohort
+
+    CohortDescriptionV2:
+      type: string
+      description: Description of the cohort
+
+    CriteriaGroupV2:
+      type: object
+      description: Group of criteria for a cohort or concept set definition
+      properties:
+        id:
+          type: string
+          description: ID of the group, immutable
+        displayName:
+          type: string
+          description: Name of the group
+        criteria:
+          type: array
+          description: Set of criteria in the group
+          items:
+            $ref: "#/components/schemas/CriteriaV2"
+        operator:
+          type: string
+          description: Operator to use when combining the criteria in the group
+          enum: ["AND", "OR"]
+        excluded:
+          type: boolean
+          description: True to exclude the group, false to include it
+
+    CriteriaV2:
+      type: object
+      description: Single criteria for a cohort or concept set definition
+      properties:
+        id:
+          type: string
+          description: ID of the criteria, immutable
+        displayName:
+          type: string
+          description: Name of the criteria
+        pluginName:
+          type: string
+          description: Name of the plugin that generated this criteria
+        selectionData:
+          type: string
+          description: Serialized plugin-specific representation of the user's selection
+        uiConfig:
+          # TODO: Remove this uiConfig property once we fully support UI plugins, because it should be identical
+          # to the static config stored in the underlay. For now, we save it separately for each criteria so we
+          # can handle schema changes in the UI code instead of in plugin-specific backend code.
+          type: string
+          description: Serialized plugin-specific UI configuration for the criteria
+
```

Comment:
Can you add back the `required` properties for all of the objects? Having them results in better generated code, for the UI at least.